holoviews/core/data/__init__.py

```diff
@@ -166,7 +171,7 @@ def __init__(self, data, **kwargs):
         initialized = Interface.initialize(type(self), data, kdims, vdims,
                                            datatype=kwargs.get('datatype'))
         (data, self.interface, dims, extra_kws) = initialized
-        super(Dataset, self).__init__(data, **dict(extra_kws, **dict(kwargs, **dims)))
+        super(Dataset, self).__init__(data, **dict(kwargs, **dict(dims, **extra_kws)))
```

Comment:
I'll have to double check this, might be a leftover from something I tried early on.